src/panel/direction/DirectionPanel.jsx

```diff
@@ -120,8 +145,15 @@ export default class DirectionPanel extends React.Component {
   computeRoutes = async () => {
     const { origin, destination, vehicle } = this.state;
     if (origin && destination) {
-      this.setState({ isDirty: false, isLoading: true, error: 0, routes: [] });
+      this.setState({
+        isDirty: false,
+        isLoading: true,
+        error: 0,
+        routes: [],
+      });
       const currentQueryId = ++this.lastQueryId;
+      fire('set_origin', origin);
+      fire('set_destination', destination);
```

Comment:
Why is this change required ?

Reply:
the idea was to show the start/end markers without waiting for the route to be computed (which can take a few seconds). If I remove these lines, and compute a route between two points clicked on the map, the last marker I clicked appears at the same time as the route, which is too late imo...

Reply:
...Do you want me to make sure any preexisting marker is cleared before that? The way I did it, one of the markers may get doubled (harmlessly)

Reply:
I am not really worried about the markers, and I agree that the markers should be visible as soon as possible.  

But a side-effect of this change is that `set_origin` and `set_destination` will call `fitMap` before the route is computed.
So selecting the origin or destination that will trigger the compute, will actually cause the map position to jump twice, before the route is visible.

Reply:
OH, indeed. That's wrong. I guess I could add a fitMap = false parameter to setOrigin/Destination... is it ok for you?

Reply:
Yes, it should be ok :+1:

Reply:
... but wait, I don't see anything that resembles fitMap() in these functions or in the functions called by them! 🤔 all they do is create a marker and add it to the map and in routeMarkers...

Reply:
actually I removed the fitMap entirely from set_origin and set_destination as there were zero cases where the fitMap was needed when calling them.
FitMap is only necessary in one case: when only one point/field is restored from the url, so I added the fitMap in this special case.